plugin/src/main/kotlin/me/tylerbwong/gradle/metalava/task/MetalavaCheckCompatibilityTask.kt

```diff
@@ -68,22 +69,23 @@ internal abstract class MetalavaCheckCompatibilityTask @Inject constructor(
             extension: MetalavaExtension,
             module: Module,
             variantName: String?
-        ): MetalavaCheckCompatibilityTask {
-            val tempFilename = project.layout.buildDirectory
-                .file(METALAVA_CURRENT_PATH).get().asFile.absolutePath
+        ): TaskProvider<MetalavaCheckCompatibilityTask> {
+            val tempFilenameProvider = project.layout.buildDirectory
+                .file(METALAVA_CURRENT_PATH).map { it.asFile.absolutePath }
             val taskName = getFullTaskName(TASK_NAME, variantName)
             val metalavaClasspath = project.getMetalavaClasspath(
                 objectFactory,
                 jarPath = extension.metalavaJarPath.get().ifEmpty { null },
                 version = extension.version.get(),
             )
-            return project.tasks.create<MetalavaCheckCompatibilityTask>(taskName) {
+            val bootClasspathProvider = project.provider { module.bootClasspath }
```

Comment:
As noted in #53, this is slow because it is scanning the java home dir recursively, looking for `rt.jar` which does not exist on Java 11.

I could address that here as well, if you want?

Reply:
After addressing this (basically looking only for `jre/lib/rt.jar` instead of scanning the entire java home), task execution time comes back down to 7 seconds, like it was previously, but this time with near-0 configuration time:
https://scans.gradle.com/s/7n2p53i6jgkiw/performance/build